blog/26-04-2023-building-a-celo-based-decentralized-medical-record-system/building-a-celo-based-decentralized-medical-record-system.md

```diff
@@ -0,0 +1,833 @@
+  ) : (
+    <button type="submit">Create Record</button>
+  )}
+</form>
+```
+
+
+With these changes, users can create and update medical records using the front end. Doctors can grant and revoke the `DOCTOR_ROLE` to other users through the smart contract methods `grantDoctorRole` and `revokeDoctorRole`. This can be done using tools like Remix or by creating additional front-end components which we will do in the next section.
+
+## Displaying Medical Records
+Now that we can create and update medical records, let's create a component that will display these records in a list. 
+- Add the following code to the `src/MedicalRecord.js` component:
+
+```javascript
+const MedicalRecordList = ({ records, onSelect }) => {
+  return (
+    <ul>
+      {records.map((record, index) => (
+        <li key={index} onClick={() => onSelect(index + 1)}>
+          {record.name} - {record.birthdate}
+        </li>
+      ))}
+    </ul>
+  );
+};
+```
+
+- Next, update the component's JSX to include the MedicalRecordList component:
+
+```jsx
+<div>
+  <h2>Your Medical Records</h2>
+  <MedicalRecordList records={records} onSelect={setSelectedRecordId} />
+</div>
+```
+
+Now, when a user clicks on a medical record in the list, the form will be populated with the data of the selected record, allowing the user to update the medical data.
+
+## Error Handling and Loading Indicators
+
+To improve the user experience, we should add error handling and loading indicators to our application. 
+- First, let's add the necessary state variables to the `src/MedicalRecord.js` component:
+
+```javascript
+const [isLoading, setIsLoading] = useState(false);
+const [error, setError] = useState(null);
+```
+
+- Next, update the `fetchRecords`, `createNewRecord`, and `updateExistingRecord` functions to handle errors and set the loading state:
+
+```javascript
+// Add error handling and loading indicator to fetchRecords
+const fetchRecords = async () => {
+  setIsLoading(true);
+  setError(null);
+  try {
+    // Fetch records code here
+  } catch (error) {
+    setError(error.message);
+  } finally {
+    setIsLoading(false);
+  }
+};
+
+// Add error handling and loading indicator to createNewRecord
+const createNewRecord = async (e) => {
+  e.preventDefault();
+  setIsLoading(true);
+  setError(null);
+  try {
+    // Create new record code here
+  } catch (error) {
+    setError(error.message);
+  } finally {
+    setIsLoading(false);
+  }
+};
+
+// Add error handling and loading indicator to updateExistingRecord
+const updateExistingRecord = async (e) => {
+  e.preventDefault();
+  setIsLoading(true);
+  setError(null);
+  try {
+    // Update existing record code here
+  } catch (error) {
+    setError(error.message);
+  } finally {
+    setIsLoading(false);
+  }
+};
+```
+
+- Finally, update the component's JSX to display the error message and loading indicator:
+
+```jsx
+{error && <p>Error: {error}</p>}
+{isLoading && <p>Loading...</p>}
+```
+
+## Testing the Application
```

Comment:
can you add a space after the first paragraph before the first header here

Reply:
done